### DISCUSSION TITLE
tinyNeoPixel single pixel update doesn't seem possible

### DISCUSSION THREAD

**Discussion by gezzerlol** [SpenceKonde/megaTinyCore#1085] [Library discussion]
I understand that this library functionality is intended to provide compatibility with the AdaFruit NeoPixel library, and I am not familiar with that.

The only LED update function in the library seems to be "show()", however, even if only a single pixel has been changed, it writes the whole array of pixels, which takes time. 

I have an application that only does single pixel updates and it could be orders of magnitude faster if I didn't have to update rest of the pixels with the same values.

Is it possible?

**Comment by grandaspanna** [comment]
It's the way those LEDs work, and not specific to the core/library. All the data goes into the first one, it strips off the first few bytes and sends the rest on, and so forth. Unless you have lots of LEDs, it's generally not an issue. You can update hundreds of them pretty quickly. If you really wanted to trim things, you might be able to send only N pixels worth of data, if you were targeting pixel N, but you'll always have to send the data for the pixels before it.

**Reply by gezzerlol** [reply]
I didn't know it worked that way. What I knew was that the first LED took the 1 LED worth of RGB data packet with index byte, substracted 1 from the index byte, and passed it to the next LED until the index byte is zero, which then it uses the RGB value to update itself in that case. If this was the case my concern would be valid.

However what you are saying is that the first LED takes whole strip of LEDs worth of RGB data, takes the first 24 bits, passes the remaining to the next LED. In this case you have to update all the LEDs, up to the LED you want to update, right?

**Reply by grandaspanna** [reply]
That's the way I understand it. The datasheet for the WS2812B has a description: https://cdn-shop.adafruit.com/datasheets/WS2812B.pdf
